cli/pack_command.go

```diff
@@ -0,0 +1,110 @@
+package cli
```

Comment:
I like the overall extraction of the command building to this new function. I am curious as to why `cli` and not `cmd`. It would appear as though the convention is more towards `cmd` and consumers would still be able to achieve the same.

Reply:
We already have a cmd package, which seemingly functions just to provide `cmd.Version`. That is then used in a bunch of places, so we can't this function in the same package as the `cmd.Version` var, because of import cycles. I can move Version to another package, but I was trying to minimize what I touched inside the code. What do you think?

Reply:
> Perhaps I'll move the Version to the pack package (so it would be pack.Version, which anyways sees clearer), and move this into cmd...

Version is a build time value, it doesn't make sense in `pack` package as that is the scope for the pack library.

I reviewed where `cmd.Version` is getting used and your statement is correct. Yet, I feel like the _right_ approach would be to propagate the version to the `Client` from `cmd` so that it's properly set in all cases, allows for configuration (other tools using pack as a library), and solves the cyclic dependency.

I'd have to investigate if there any assumptions being made about the `Client` name and version but I believe it was solely introduced for traceability/compliance.

Reply:
> Version is a build time value that applies to the cli only, it doesn't make sense in pack package as that is the scope for the pack library.

We can discuss whether this makes sense to change and possible hardcode the value instead of providing it in at build time in which case it would be accurate for both cli and lib use cases.

Reply:
Found the original feature request: https://github.com/buildpacks/pack/issues/261

Reply:
TLDR: Moving it to the top-level `pack` package makes sense and instead of using some external build-time value we should hardcode it (can be done later).

> 1. know what metadata to expect on the builder
> 1. know the organizational structure of the image FS (Example: lifecycle binary paths)
> 1. know whether a builder is compatible to run with another version of pack.
> 1. can provide this information in bug reports involving a custom builder
> 1. can satisfy their curiosity

Based on this information:

1. Builders should be an extension spec instead of relying on pack version (@ekcasey 😄)
2. We should hardcode the value so that the pack version is also applied when used as a lib.

Sorry for the long windedanswerr @dfreilich.

Reply:
Interesting. I know that @sclevine had been a fan of build-time versions, so I wonder what he thinks.

Reply:
But, to clarify, you think I should move the `Version` to the `pack` package, and then move the `NewPackCommand(...)` to `cmd`?

Reply:
@jromero I assumed that was what you meant, and went ahead with it (I also had to inject the version into the builders Version method, in order to prevent cycles). Let me know what you think!